src/Tests/DependencyInjectionExtensions_UnitTest.cs

```diff
@@ -31,8 +31,10 @@ public async Task BeAbleToInstantiateAllRegisteredTypes()
 
             foreach (var serviceDescriptor in services)
             {
-                serviceDescriptor.ServiceType.Should().NotBeNull();
-                serviceDescriptor.ServiceType.FullName.Should().NotBeNullOrEmpty();
+                if (serviceDescriptor.ServiceType == null)
```

Comment:
The compiler is dumb and doesn't understand fluent assertions so the guard clause and throwing exceptions is easier, the test still fails.

Reply:
If I recall correctly, it doesn't understand XUnit's standard `Assert` checks either.